### ISSUE TITLE
AUTN Validation MAC mismatch

### ISSUE THREAD

**Issue by e99x** [aligungr/UERANSIM#260]
Hello.

I use Free5gc(v3.0.3).

I want to connect UERANSIM, but I get the following error.
---------------------------------------------------------------------
UERANSIM v3.1.1
[2021-02-26 00:49:47.982] [nas] [debug] NAS layer started
[2021-02-26 00:49:47.982] [rrc] [debug] RRC layer started
[2021-02-26 00:49:47.982] [nas] [info] UE switches to state: MM-DEREGISTERED/PLMN-SEARCH
[2021-02-26 00:49:47.983] [nas] [info] UE connected to gNB
[2021-02-26 00:49:47.983] [nas] [info] UE switches to state: MM-DEREGISTERED/NORMAL-SERVICE
[2021-02-26 00:49:47.983] [nas] [info] UE switches to state: MM-REGISTERED-INITIATED/NA
[2021-02-26 00:49:47.983] [rrc] [debug] Sending RRC Setup Request
[2021-02-26 00:49:47.983] [rrc] [info] RRC connection established
[2021-02-26 00:49:47.983] [nas] [info] UE switches to state: CM-CONNECTED
[2021-02-26 00:49:48.004] [nas] [debug] Received rand[935F5E4DBBA731DFB16121147FF999C5] autn[22AD6A98C2098000DDD62FCE6604755D]
[2021-02-26 00:49:48.004] [nas] [debug] Calculated res[D5788CA2C958AFE2] ck[F7FEF4D1446F203F643201E55DC4765E] ik[7A7C16D820256BED1D8FD8A0094C3EC9] ak[85251EFF8C0A] mac_a[D2D1A9FB1BC44024]
[2021-02-26 00:49:48.004] [nas] [debug] Used snn[5G:mnc093.mcc208.3gppnetwork.org] sqn[A78874674E03]
[2021-02-26 00:49:48.004] [nas] [error] AUTN validation MAC mismatch. expected: D2D1A9FB1BC44024 received: DDD62FCE6604755D
[2021-02-26 00:49:48.006] [nas] [error] Authentication Reject received.

---------------------------------------------------------------------

I've tried setting IA3 to false, but it doesn't work.

Is there any other solution and Is there anything I should check?

**Comment by ananay** [comment]
Same error here. Please help!

**Comment by ananay** [comment]
Hey! I figured out the issue. What you gotta do is go to the Free5GC web console and change Operator Code Type from OPc and to OP. (that was the case for me and changing that worked).

![image](https://user-images.githubusercontent.com/5569219/109300582-e9e83680-77f3-11eb-8449-a4271130d53f.png)

**Comment by aligungr** [comment]
@ananay Yes, many people are confused about this. I think, I should change the sample UE config file and set op type to 'OPC', because in free5gc web console OPc is selected by default.

@e99x Please check the following paramters in UERANSIM config yaml, they should match exactly between UE and core network:
- supi,
- mcc
- mnc
- key
- op
- opType
- amf

**Comment by e99x** [comment]
Hello！ ananay and aligungr

I tried it and it was solved.
Thank you.
